Web/Edubase.Web.UI/Controllers/Api/AcademyOpeningsApiController.cs

```diff
@@ -40,48 +40,60 @@ public class AcademyOpeningsApiController : ApiController
         /// <param name="to"></param>
         /// <param name="skip"></param>
         /// <param name="take"></param>
+        /// <param name="establishmentTypeId"></param>
         /// <returns></returns>
-        [Route("api/academy-openings/list/{from:datetime}/{to:datetime}/{skip:int}/{take:int}"), HttpGet]
-        public async Task<dynamic> GetListAsync(DateTime from, DateTime to, int skip, int take)
+        [Route(
+             "api/academy-openings/list/{from:datetime}/{to:datetime}" +
+             "/{skip:int}/{take:int}/{establishmentTypeId?}"),
+         HttpGet]
+        public async Task<dynamic> GetListAsync(DateTime from, DateTime to, int skip, int take,
+            string establishmentTypeId = null)
         {
+            if (!AcademyUtility.DoesHaveAccessAuthorization(User, establishmentTypeId))
+                throw AcademyUtility.GetAccessViolationException();
```

Comment:
Looking elsewhere in the codebase, it seems that PermissionDeniedException might be the one to throw